### PR TITLE
Add Symfony 5 and Twig 3 support

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is needed to keep moving forward SonataAdminBundle major release and compat with Symfony 5.0.

Closes #305 .

## Changelog

```markdown
### Added
- Added support for Symfony 5
- Added support for Twig 3
```

### CODE REVIEW

File: src/Templating/Helper/BaseHelper.php [hunk 1]
```diff
@@ -28,8 +27,13 @@
  *
  * @author Alexander <iam.asm89@gmail.com>
  */
-abstract class BaseHelper extends Helper
+abstract class BaseHelper
```
Comment (jordisala1991) [inline src/Templating/Helper/BaseHelper.php:30]: WDYT about this BC break @sonata-project/contributors . The methods added by this extends are on this class now, so it is only a BC break for those checking what classes does this extend or implemented interface.

IMO if everything works, it is fine. That helper does not bring anything, it is just a get/set

Comment (jordisala1991) [reply src/Templating/Helper/BaseHelper.php]: NVM, let's add first support for SF 5 and twig 3, without BC breaks. This is not needed.

File: phpstan-baseline.neon [hunk 2]
```diff
@@ -0,0 +1,6 @@
+parameters:
+	ignoreErrors:
+		-
+			message: "#^Class Sonata\\\\UserBundle\\\\Model\\\\User not found\\.$#"
```
Comment (VincentLanglet) [inline phpstan-baseline.neon:4]: What is this new ?

Comment (VincentLanglet) [reply phpstan-baseline.neon:4]: Oh, I see, you removed the `"sonata-project/user-bundle": "^3.6 || ^4.0"` dependency.
Should we keep it then ?

Comment (jordisala1991) [reply phpstan-baseline.neon:4]: We should keep it yes. The integration with the user-bundle is deprecated. I just removed the required-dev dependency in order to be able to test with Symfony 5.0

**Comment by phansys** [comment]
I'd like to have at least a job checking the deprecated integration with sonata-project/user-bundle. Otherwise we could break that without any notice.

**Comment by jordisala1991** [comment]
Added the necessary code here, and also dev-kit PR: https://github.com/sonata-project/dev-kit/pull/796 @phansys

File: composer.json [hunk 3]
```diff
@@ -22,23 +22,22 @@
         }
     ],
     "require": {
-        "php": "^7.1",
-        "symfony/config": "^4.4",
-        "symfony/dependency-injection": "^4.4",
-        "symfony/http-foundation": "^4.4",
-        "symfony/http-kernel": "^4.4",
+        "php": "^7.2",
+        "symfony/config": "^4.4 || ^5.0",
```
Comment (core23) [inline composer.json:26]: Symfony 5.0 is unsupported: https://symfony.com/releases/5.0

Comment (jordisala1991) [reply composer.json]: Fixed

**Comment by VincentLanglet** [comment]
@jordisala1991 I just saw you let `"symfony/templating": "^4.4",`

**Comment by jordisala1991** [comment]
yes, that is on purpose. Symfony templating does not exist on 5.0.

the ideal solution would be remove it, but that can only be safely implement on master because it implies a bc break here.

**Pull request merged**